File: tests/detectors/samples/S03/typescript/negative.ts
export async function fetchPage(url: string) {
  const res = await fetch(url, { signal: AbortSignal.timeout(5_000) });
  if (res.status === 429) {
    const header = res.headers.get("Retry-After");
    const waitMs = header ? Number(header) * 1000 : 60_000;
    await new Promise(r => setTimeout(r, waitMs));
    return fetchPage(url);
  }
  return res;
}

File: tests/detectors/samples/S03/python/negative.py
import requests, time

def fetch_page(url):
    r = requests.get(url, timeout=5)
    if r.status_code == 429:
        retry_after = r.headers.get("Retry-After")
        time.sleep(int(retry_after) if retry_after else 60)
        return fetch_page(url)
    return r
